contrib/vcloud/benchmarkclient_executor.py

```diff
@@ -0,0 +1,449 @@
+                "Missing file '{0}', cannot run benchmark within cloud.".format(
+                    os.path.normpath(file)
+                )
+            )
+        if os.path.isdir(file) and not os.listdir(file):
+            # VCloud can not handle empty directories, lets ignore them
+            logging.warning(
+                "Empty directory '%s', ignoring directory for cloud execution.",
+                os.path.normpath(file),
+            )
+        else:
+            validToolpaths.add(file)
+
+    return (workingDir, validToolpaths)
+
+
+def handleCloudResults(benchmark, output_handler, start_time, end_time):
+
+    outputDir = benchmark.log_folder
+    if not os.path.isdir(outputDir) or not os.listdir(outputDir):
+        # outputDir does not exist or is empty
+        logging.warning(
+            "Cloud produced no results. Output-directory is missing or empty: %s",
+            outputDir,
+        )
+
+    # Write worker host informations in xml
+    parseAndSetCloudWorkerHostInformation(outputDir, output_handler, benchmark)
+
+    if start_time and end_time:
+        usedWallTime = (end_time - start_time).total_seconds()
+    else:
+        usedWallTime = None
+
+    # write results in runs and handle output after all runs are done
+    executedAllRuns = True
+    runsProducedErrorOutput = False
+    for runSet in benchmark.run_sets:
+        if not runSet.should_be_executed():
+            output_handler.output_for_skipping_run_set(runSet)
+            continue
+
+        output_handler.output_before_run_set(runSet, start_time=start_time)
+
+        for run in runSet.runs:
+            dataFile = run.log_file + ".data"
+            if os.path.exists(dataFile) and os.path.exists(run.log_file):
+                try:
+                    values = parseCloudRunResultFile(dataFile)
+                    if not benchmark.config.debug:
+                        os.remove(dataFile)
+                except IOError as e:
+                    logging.warning(
+                        "Cannot extract measured values from output for file %s: %s",
+                        run.identifier,
+                        e,
+                    )
+                    output_handler.all_created_files.add(dataFile)
+                    output_handler.set_error("missing results", runSet)
+                    executedAllRuns = False
+                else:
+                    output_handler.output_before_run(run)
+                    run.set_result(values, ["host"])
+                    output_handler.output_after_run(run)
+            else:
+                logging.warning("No results exist for file %s.", run.identifier)
+                output_handler.set_error("missing results", runSet)
+                executedAllRuns = False
+
+            if os.path.exists(run.log_file + ".stdError"):
+                runsProducedErrorOutput = True
+
+            # The directory structure differs between direct and webclient mode when using VCloud.
```

Comment:
This comment also seems outdated, there should not need to be a reference to the WebClient in this script. Instead, it should make the result look like the result from plain BenchExec.

Reply:
I think I have mentioned it in the PR comment that I have tried to keep stuff as it is because it makes the maintenance of  code duplication simple.

But if you prefer to keep two slightly different files for the `benchmarkclient_executor` then so be it.

Reply:
No, of course as many files as possible should be identical.

But still, there is no reason to adjust something to the way the webclient mode does something. It should be adjusted to the way BenchExec does things.

Reply:
updated the comment